src/domain/app/apollo/apolloClient.ts

```diff
@@ -436,62 +436,87 @@ const MUTATIONS_NOT_TO_SHOW_FORBIDDEN_ERROR = [
   'UpdatePlace',
 ];
 
-const errorLink = onError(({ graphQLErrors, networkError, operation }) => {
-  const isMutation = Boolean(
-    operation.query.definitions.find(
-      (definition) =>
-        definition.kind === 'OperationDefinition' &&
-        definition.operation === 'mutation'
-    )
-  );
-  if (graphQLErrors) {
-    graphQLErrors.forEach(({ message, locations, path }) => {
-      const errorMessage = `[GraphQL error]: Message: ${message}, Location: ${locations}, Path: ${path}`;
-      Sentry.captureMessage(errorMessage);
-    });
-  }
+const createErrorLink = ({
+  addNotification,
+}: {
+  addNotification: (props: NotificationProps) => void;
+}) =>
+  onError(({ graphQLErrors, networkError, operation }) => {
+    const isMutation = Boolean(
+      operation.query.definitions.find(
+        (definition) =>
+          definition.kind === 'OperationDefinition' &&
+          definition.operation === 'mutation'
+      )
+    );
+    if (graphQLErrors) {
+      graphQLErrors.forEach(({ message, locations, path }) => {
+        const errorMessage = `[GraphQL error]: Message: ${message}, Location: ${locations}, Path: ${path}`;
+        Sentry.captureMessage(errorMessage);
+      });
+    }
 
-  if (
-    (isMutation || QUERIES_TO_SHOW_ERROR.includes(operation.operationName)) &&
-    networkError
-  ) {
-    switch ((networkError as ServerError).statusCode) {
-      case 400:
-        if (
-          !MUTATIONS_NOT_TO_SHOW_VALIDATION_ERROR.includes(
-            operation.operationName
-          )
-        ) {
-          toast.error(getValue(i18n.t('errors.validationError'), ''));
-        }
-        break;
-      case 401:
-        toast.error(getValue(i18n.t('errors.authorizationRequired'), ''));
-        break;
-      case 403:
-        if (
-          !MUTATIONS_NOT_TO_SHOW_FORBIDDEN_ERROR.includes(
-            operation.operationName
-          )
-        ) {
-          toast.error(getValue(i18n.t('errors.forbidden'), ''));
-        }
-        break;
-      case 404:
-        toast.error(getValue(i18n.t('errors.notFound'), ''));
-        break;
-      case 410:
-        toast.error(getValue(i18n.t('errors.deleted'), ''));
-        break;
-      default:
-        if (
-          !MUTATIONS_NOT_TO_SHOW_SERVER_ERROR.includes(operation.operationName)
-        ) {
-          toast.error(getValue(i18n.t('errors.serverError'), ''));
-        }
+    if (
```

Comment:
SonarCloud scan results are mentioning high cognitive complexity for this function, and it's coming from the nested conditions in this part of the function's code. It seems that the new changes haven't really caused the code smell issue, but I would suggest refactoring this still a bit.

Some suggestions:
- do an early return before the current switch statement instead of nesting the switch statement inside the `if` condition
- use an object literal instead of the switch statement (https://www.30secondsofcode.org/js/s/switch-object/)
- or instead of an object literal: split the switch statement cases into separate functions

Reply:
I will split this PR to several smaller PRs to make it easier to review. This part is already fixed in https://github.com/City-of-Helsinki/linkedcomponents-ui/pull/240. I will add missing PRs later today.